### PR TITLE
Implement the complement to checkAccess

### PR DESCRIPTION
**Pull request by kjetilk**
As previously mentioned, to allow for the reasons for rejection to be communicated to the client as wanted in https://github.com/solid/node-solid-server/issues/468 , and maintain a single responsibility module, I suggested in #2  that the easiest was to implement the complement to `checkAccess`, i.e. `accessDenied`. This patch does that. It also adds quite a few more tests, and checks some more corner cases, one that caused me a some pain. 

Please poke!

### CODE REVIEW

File: test/unit/access-denied-test.js [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+  var result
+  const store = $rdf.graph()
+  /*
+  let ACLtext = prefixes + ` <#auth> a acl:Authorization;
+    acl:mode acl:Read;
+    acl:agent bob:me;
+    acl:accessTo <${file1.uri}> .
+    `
+  $rdf.parse(ACLtext, store, containerAcl.uri, 'text/turtle')
+*/
```
Comment (RubenVerborgh) [inline test/unit/access-denied-test.js]: commented?

Comment (kjetilk) [reply test/unit/access-denied-test.js]: Yeah, actually, many of the tests were taken from the check-access-test, and this was commented out there. I guess I should remove the outcommented stuff.
